pkg/controller/statusmanager/status_manager.go

```diff
@@ -47,7 +48,7 @@ func New(client client.Client, name, version string) *StatusManager {
 }
 
 // Set updates the ClusterOperator.Status with the provided conditions
-func (status *StatusManager) Set(conditions ...configv1.ClusterOperatorStatusCondition) {
+func (status *StatusManager) Set(reachedAvailableLevel bool, conditions ...configv1.ClusterOperatorStatusCondition) {
```

Comment:
I'm surprised that we can't compute `reachedAvailableLevel` directly from the conditions - is that really true?